### PR TITLE
Add array input function unit test

### PR DESCRIPTION
**Pull request by Fladolcetta**
[x] This contribution adheres to [CONTRIBUTING.md]

### What does this Pull Request accomplish?

Adds additional unit testing to verify that calling functions with input arrays works as expected

### What testing has been done?

Ran builds, unit tests and flake8

### CODE REVIEW

File: generated/nifake/session.py [hunk 1]
```diff
@@ -316,6 +316,19 @@ def _abort(self):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    def array_input_function(self, buffer_size, a_array):
```
Comment (marcoskirsch) [inline generated/nifake/session.py]: should be an_array ?

File: src/nifake/metadata/functions.py [hunk 2]
```diff
@@ -1248,15 +1248,49 @@
...
+                'name': 'vi',
+                'type': 'ViSession',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi**',
+                },
+            },
+            {
+                'direction': 'in',
+                'enum': None,
+                'name': 'bufferSize',
```
Comment (marcoskirsch) [inline src/nifake/metadata/functions.py]: rename to size or numberOfElements
(buffer size is in bytes usually)

File: src/nifake/metadata/functions.py [hunk 3]
```diff
@@ -1248,15 +1248,49 @@
...
+                'type': 'ViInt32',
+                'documentation': {
+                    'description': 'Buffersize of the array.',
```
Comment (marcoskirsch) [inline src/nifake/metadata/functions.py]: Buffersize is not a word, also, it's incorrect. This is the number of elements.

File: src/nifake/metadata/functions.py [hunk 4]
```diff
@@ -1248,15 +1248,49 @@
...
+                'is_buffer': True,
+                'name': 'aArray',
+                'type': 'ViReal64',
+                'documentation': {
+                    'description': 'Contains an array of float numbers',
+                },
+            },
+        ],
+        'documentation': {
+            'description': 'This function takes one parameter other than the session.',
```
Comment (marcoskirsch) [inline src/nifake/metadata/functions.py]: fix

File: generated/nifake/session.py (hunk 1 above)
Comment (Fladolcetta) [reply generated/nifake/session.py]: fixed

File: src/nifake/metadata/functions.py (hunk 2 above)
Comment (Fladolcetta) [reply src/nifake/metadata/functions.py]: fixed

File: src/nifake/metadata/functions.py (hunk 4 above)
Comment (Fladolcetta) [reply src/nifake/metadata/functions.py]: Fixed

**Pull request merged**